datafusion/expr-common/src/groups_accumulator.rs

```diff
@@ -143,6 +145,25 @@ pub trait GroupsAccumulator: Send {
     /// [`Accumulator::state`]: crate::accumulator::Accumulator::state
     fn state(&mut self, emit_to: EmitTo) -> Result<Vec<ArrayRef>>;
 
+    /// Returns `true` if this accumulator supports blocked mode.
+    fn supports_blocked_mode(&self) -> bool {
+        false
+    }
+
+    /// Switch the accumulator to flat or blocked mode.
+    /// You can see detail about the mode on [GroupStatesMode].
+    ///
+    /// After switching mode, all data in previous mode will be cleared.
+    fn switch_to_mode(&mut self, mode: GroupStatesMode) -> Result<()> {
```

Comment:
I understand why you introduced this API, but I think it makes the accumulators harder to reason about because now each now has two potential modes so there are two similar, but not the same parallel implementations that we have to ensure are tested. I had an idea to avoid this switch_to_mode API below

Reply:
:thinking: But as I think, the similar function may be needed now as I memtioned above avout `The need of switching from hash aggr to streaming aggr` problem.

Maybe we can support a similar function such as `alter_block_size`?